src/opm/parser/eclipse/EclipseState/Schedule/Group/Group.cpp

```diff
@@ -79,19 +78,21 @@ namespace Opm {
     }
 
     bool Group::isProductionGroup(size_t timeStep) const {
-        return bool( m_isProductionGroup.get(timeStep) );
+      return ((this->group_type.get(timeStep) & GroupType::PRODUCTION) == GroupType::PRODUCTION);
```

Comment:
I think I'd prefer if this were written in terms of a helper function, e.g.,
```C++
return this->groupBitIsSet(timeStep, GroupType::PRODUCTION);
```
in which `groupBitIsSet()` (or some other name) would be the obvious
```C++
bool groupBitIsSet(const std::size_t timeStep, const GroupType gtype) const
{
    return (this->group_type.get(timeStep) & gtype) == gtype;
}
```
Then you could reuse that for `isInjectionGroup()` too in order to reduce the amount of bit manipulation.